core/src/main/java/org/apache/carbondata/core/util/path/CarbonTablePath.java

```diff
@@ -274,6 +275,35 @@ public String getCarbonIndexFilePath(final String taskId, final String partition
           + "]");
     }
   }
+
+  /**
+   * Below method will be used to get the carbon index file path
+   * @param taskId
+   *        task id
+   * @param partitionId
+   *        partition id
+   * @param segmentId
+   *        segment id
+   * @param bucketNumber
+   *        bucket number
+   * @param timeStamp
+   *        timestamp
+   * @return carbon index file path
+   */
+  public String getCarbonIndexFilePath(String taskId, String partitionId, String segmentId,
+      String bucketNumber, String timeStamp, ColumnarFormatVersion columnarFormatVersion) {
+    if (ColumnarFormatVersion.V3 != columnarFormatVersion) {
```

Comment:
Better add a switch case and make V3 as default instead of strictly checking with V3 format.